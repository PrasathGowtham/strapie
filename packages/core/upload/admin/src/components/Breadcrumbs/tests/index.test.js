import React from 'react';
import { render as renderTL, screen, fireEvent, waitFor } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { Breadcrumbs } from '../index';

jest.mock('../../../hooks/useFolderStructure');

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useQueryParams: jest.fn().mockReturnValue([{ query: { folder: 22 } }]),
}));

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnWindowFocus: false,
    },
  },
});

const defaultBreadcrumbs = [
  {
    href: '/',
    id: null,
    label: 'Media Library',
  },
  [],
  { href: '/', id: 21, label: 'parent folder' },
  { id: 22, label: 'current folder' },
];

const setup = (props) =>
  renderTL(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <IntlProvider locale="en" messages={{}}>
          <ThemeProvider theme={lightTheme}>
            <Breadcrumbs breadcrumbs={defaultBreadcrumbs} label="Navigation" as="nav" {...props} />
          </ThemeProvider>
        </IntlProvider>
      </MemoryRouter>
    </QueryClientProvider>
  );

describe('Media Library | Breadcrumbs', () => {
  test('should render and match snapshot', () => {
    const { container } = setup({ currentFolderId: 22 });

    expect(container.querySelector('nav')).toBeInTheDocument();
    expect(screen.getByText('parent folder')).toBeInTheDocument();
    expect(screen.getByText('current folder')).toBeInTheDocument();
    expect(screen.getByText('Media Library')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('should store other ascendants in simple menu', async () => {
    const { getByRole } = setup({ currentFolderId: 22 });

    const simpleMenuButton = getByRole('button', { name: /get more ascendants folders/i });
    fireEvent.mouseDown(simpleMenuButton);

    await waitFor(() => {
      expect(screen.getByText('second child')).toBeInTheDocument();
    });
  });
});
