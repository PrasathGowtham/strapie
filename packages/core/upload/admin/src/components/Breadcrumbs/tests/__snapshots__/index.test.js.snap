// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Library | Breadcrumbs should render and match snapshot 1`] = `
.c16 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  padding-right: 4px;
  padding-left: 4px;
}

.c12 {
  padding-left: 8px;
}

.c14 {
  padding-top: 4px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
}

.c7 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c11 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: 1.14;
  color: #32324d;
}

.c15 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2:first-child {
  margin-left: calc(-1*8px);
}

.c4 {
  border-radius: 4px;
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.43;
  padding: 4px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover,
.c4:focus {
  background-color: #dcdce4;
  color: #4a4a6a;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c8 svg {
  height: 12px;
  width: 12px;
}

.c8 svg > g,
.c8 svg path {
  fill: #ffffff;
}

.c8[aria-disabled='true'] {
  pointer-events: none;
}

.c8:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c8:focus-visible {
  outline: none;
}

.c8:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4945ff;
  border: 1px solid #4945ff;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid transparent;
  background: transparent;
}

.c9 .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 .c6 {
  color: #ffffff;
}

.c9[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c9[aria-disabled='true'] .c6 {
  color: #666687;
}

.c9[aria-disabled='true'] svg > g,.c9[aria-disabled='true'] svg path {
  fill: #666687;
}

.c9[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c9[aria-disabled='true']:active .c6 {
  color: #666687;
}

.c9[aria-disabled='true']:active svg > g,.c9[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c9:hover {
  background-color: #f6f6f9;
}

.c9:active {
  border: 1px solid undefined;
  background: undefined;
}

.c9 .c6 {
  color: #32324d;
}

.c9 svg > g,
.c9 svg path {
  fill: #8e8ea9;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 svg {
  height: 4px;
  width: 6px;
}

.c10 {
  padding: 4px 12px;
}

.c10:hover,
.c10:focus {
  background-color: #dcdce4;
}

<div>
  <nav
    aria-label="Navigation"
    class="c0 "
  >
    <ol
      class="c0 c1 c2"
    >
      <li
        class="c0 c3"
      >
        <a
          aria-current="page"
          class="c4 active"
          href="/"
        >
          Media Library
        </a>
        <div
          aria-hidden="true"
          class="c0 c5"
        >
          <span
            class="c6 c7"
          >
            /
          </span>
        </div>
      </li>
      <li
        class="c0 c3"
      >
        <div>
          <button
            aria-controls="simplemenu-1"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Get more ascendants folders"
            class="c8 c9 c10"
            label="..."
            type="button"
          >
            <span
              class="c6 c11"
            >
              ...
            </span>
            <div
              aria-hidden="true"
              class="c0 c12"
            >
              <span
                class="c13"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  height="1em"
                  viewBox="0 0 14 8"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                    fill="#32324D"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </button>
        </div>
        <div
          aria-hidden="true"
          class="c0 c5"
        >
          <span
            class="c6 c7"
          >
            /
          </span>
        </div>
      </li>
      <li
        class="c0 c3"
      >
        <a
          aria-current="page"
          class="c4 active"
          href="/"
        >
          parent folder
        </a>
        <div
          aria-hidden="true"
          class="c0 c5"
        >
          <span
            class="c6 c7"
          >
            /
          </span>
        </div>
      </li>
      <li
        class="c0 c3"
      >
        <div
          class="c0 c14"
        >
          <span
            aria-current="true"
            class="c6 c15"
          >
            current folder
          </span>
        </div>
      </li>
    </ol>
  </nav>
  <div
    class="c16"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
